Error:  git rev-parse HEAD exited with a status of 128.

Hey friends,

Thanks for providing this awesome continuous releases tool!

Since yesterday, I have been running in the following errors when trying to publish a package ([here is the action run](https://github.com/marcoroth/herb/actions/runs/22510318329/job/65218334936#step:4:1))

```
Run npx pkg-pr-new publish --compact --comment=update --bin --no-template './javascript/packages/formatter' './javascript/packages/linter' './javascript/packages/language-server'
  npx pkg-pr-new publish --compact --comment=update --bin --no-template './javascript/packages/formatter' './javascript/packages/linter' './javascript/packages/language-server'
  shell: /usr/bin/bash -e {0}
npm warn exec The following package was not found and will be installed: pkg-pr-new@0.0.65
Error:  git rev-parse HEAD exited with a status of 128.

fatal: not a git repository (or any of the parent directories): .git
  
  fatal: not a git repository (or any of the parent directories): .git
  at normalizeResult (/home/runner/.npm/_npx/e8f433c388b218fc/node_modules/@jsdevtools/ez-spawn/lib/normalize-result.js:31:11)
  at ChildProcess.<anonymous> (/home/runner/.npm/_npx/e8f433c388b218fc/node_modules/@jsdevtools/ez-spawn/lib/async.js:79:19)
  at ChildProcess.emit (node:events:524:28)
  at ChildProcess._handle.onexit (node:internal/child_process:293:12) 

Error:  git rev-parse HEAD exited with a status of 128.

fatal: not a git repository (or any of the parent directories): .git
Error: Process completed with exit code 1.
```

It looks like this was introduced in #470

There is a good chance that I have been "abusing" the fact that it didn't need a git repository present to publish a package, but I thought it might be worth reporting anyway.

Would it make sense to skip that commit sha parsing if no git repository is found?

Thank you!

--- a/packages/app/e2e.test.ts
+++ b/packages/app/e2e.test.ts
@@ -18,6 +18,13 @@ let tempDir: string;
 let githubOutputPath: string;
 
 let worker: UnstableDevWorker;
+
+const { stdout: gitRevParseOutput } = await ezSpawn.async(
+  "git rev-parse HEAD",
+  { stdio: "overlapped" },
+);
+const gitRevParse = gitRevParseOutput.trim();
+
 beforeAll(async () => {
   tempDir = await fs.mkdtemp(path.join(os.tmpdir(), E2E_TEMP_DIR_PREFIX));
 
@@ -122,6 +129,7 @@ describe.sequential.each([
       GITHUB_SHA: payload.workflow_run.head_sha,
       GITHUB_ACTION: payload.workflow_run.id,
       GITHUB_JOB: payload.workflow_run.name,
+      GITHUB_EVENT_NAME: payload.workflow_run.event,
       GITHUB_REF_NAME: pr
         ? `${pr.payload.number}/merge`
         : payload.workflow_run.head_branch,
@@ -150,10 +158,8 @@ describe.sequential.each([
 
   it(`serves and installs playground-a for ${mode}`, async () => {
     const [owner, repo] = payload.repository.full_name.split("/");
-    const { stdout: gitHeadSha } = await ezSpawn.async("git rev-parse HEAD", {
-      stdio: "overlapped",
-    });
-    const sha = gitHeadSha.trim().substring(0, 7);
+    const fullSha = pr ? payload.workflow_run.head_sha : gitRevParse;
+    const sha = fullSha.substring(0, 7);
     const ref = pr?.payload.number ?? payload.workflow_run.head_branch;
 
     // Test download with SHA
@@ -193,10 +199,8 @@ describe.sequential.each([
 
   it(`serves and installs playground-b for ${mode}`, async () => {
     const [owner, repo] = payload.repository.full_name.split("/");
-    const { stdout: gitHeadSha } = await ezSpawn.async("git rev-parse HEAD", {
-      stdio: "overlapped",
-    });
-    const sha = gitHeadSha.trim().substring(0, 7);
+    const fullSha = pr ? payload.workflow_run.head_sha : gitRevParse;
+    const sha = fullSha.substring(0, 7);
 
     // Test download
     const response = await worker.fetch(

--- a/packages/cli/index.ts
+++ b/packages/cli/index.ts
@@ -233,13 +233,24 @@ const main = defineCommand({
             process.exit(1);
           }
 
-          const { stdout: gitShaOutput } = await ezSpawn.async(
-            "git rev-parse HEAD",
-            {
-              stdio: "overlapped",
-            },
-          );
-          const sha = gitShaOutput.trim();
+          let { sha } = await checkResponse.json();
+
+          // on pull_request events, GitHub creates a virtual merge commit that doesn't
+          // actually belong to the PR, so we use the workflow's head_sha instead
+          if (process.env.GITHUB_EVENT_NAME !== "pull_request") {
+            try {
+              const { stdout: gitRevParseOutput } = await ezSpawn.async(
+                "git rev-parse HEAD",
+                { stdio: "overlapped" },
+              );
+
+              sha = gitRevParseOutput.trim();
+            } catch {
+              // git rev-parse fails when the CLI runs outside a git repository
+              // (e.g. the checkout was done in a separate job and only artifacts were restored here)
+              // falling back to the workflow's head_sha from checkResponse
+            }
+          }
 
           const deps: Map<string, string> = new Map(); // pkg.pr.new versions of the package
           const realDeps: Map<string, string> | null = isPeerDepsEnabled

